bloom-housing/bloom
# feat: upgrade react to 18

# Pull Request Template

## Issue Overview

This PR addresses #3313 #3347 #3366 


## Description

This PR does the following changes:
- Upgrade React from 17 to 18. This includes pulling in the latest version of UI-Components
- Updated all instances of `data-test-id` to `data-testid`. This standardizes everything between cypress and our unit tests
- Upgrades Cypress from 9 to 12. This includes the following required changes
  - Directory changes from `integration` to `e2e`
  - Tests no longer can depend on previous tests so all tests that relied on things such as being logged in have to now do per test. Most just needed to change "before" to "beforeEach", but some also had to re-navigate to the correct page
- The mapbox address endpoints keeps returning a suggested address even with the junk address typed. Because of this I migrated that api call to a fixture in our cypress test. This will make the test more consistent as that is one of our flakiest test and also reduce our total calls to our mapbox api

## How Can This Be Tested/Reviewed?

This change will require testing of every page to make sure no warnings appear. I'd recommend pulling it down, reseeding, and checking every page as best you can. 

Also the UI is new for the Cypress GUI. It appears to be faster for me. However, there is now no "run all" button so if you want to test the cypress test you will need to run each desired test individually.


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
